packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java

```diff
@@ -342,79 +293,184 @@ public void onCancel(Object arguments) {
               @Override
               public void run() {
                 cameraPermissionContinuation = null;
-                openCamera(result);
+                if (!hasCameraPermission()) {
+                  if (result != null) {
+                    result.error(
+                        "cameraPermission", "MediaRecorderCamera permission not granted", null);
+                    return;
+                  }
```

Comment:
If result == null we should send a message on the eventSink

Reply:
Actually result cannot be null here, I removed the null check.

Reply:
Most of the time *result* can't be null, but there is one case (the *open* method) where it can. I added annotations to clarify that.